src/gui/sdlmain.cpp

```diff
@@ -4622,6 +4636,25 @@ int sdl_main(int argc, char *argv[])
 		sdl_version.major, sdl_version.minor, sdl_version.patch,
 		SDL_GetCurrentVideoDriver(), SDL_GetCurrentAudioDriver());
 
+	// Write the default primary config if it doesn't exist when not in secure
+	// mode.
+	if (!arguments->securemode) {
+		const auto primary_config_path = SETUP_GetPrimaryConfigPath();
+
+		if (!path_exists(primary_config_path)) {
+			// No config is loaded at this point, so we're writing the default
+			// settings to the primary config.
+			const auto path_string = primary_config_path.string();
+			if (control->WriteConfig(path_string)) {
```

Comment:
WriteConfig should take in `primary_config_path` now that you've changed it to take a `std_fs::path`.  Otherwise it gets implicitly converted from the string.

Reply:
Not on Windows on MSVC & co. I had to add it back for those because I got all Windows targets failing...

Reply:
Oh... ok. I think you're right... really need some sleep...

Reply:
On Windows, it's a problem when you have a `std_fs::path` and want a C string.  Doing `path.c_str()` can cause compile errors because it returns a UTF16 string.  In that case, you need to do `path.string().c_str()` to make it UTF8.  I've ran into that same thing before in Dosbox and I see you did just that in the call to `fopen`

Here, it's fine to just pass along the path.  Otherwise you're converting to string then back to path (implicitly) for no reason.

Reply:
Yeah, thanks, I realised eventually, I was just really tired 😄 I'll fix it shortly.

Reply:
Done.